### PR TITLE
[cloud](point query) support short circuit query in cloud mode

### PR DESCRIPTION
**Pull request by eldenmoon**
Add version for cloud rowset version

In cloud mode, rs_version_map does not get updated after a successful load. We have to sync rowset version before the point query.
Add a new session variable enable_snapshot_point_query to control whether enable or not.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/service/point_query_executor.cpp [hunk 1]
```diff
@@ -191,14 +194,16 @@ Status PointQueryExecutor::init(const PTabletKeyLookupRequest* request,
             RETURN_IF_ERROR(reusable_ptr->init(t_desc_tbl, t_output_exprs.exprs, 1));
         }
     }
-    // TODO(plat1ko): CloudStorageEngine
-    _tablet = ExecEnv::GetInstance()->storage_engine().to_local().tablet_manager()->get_tablet(
-            request->tablet_id());
+    auto res = ExecEnv::get_tablet(request->tablet_id());
+    _tablet = !res.has_value() ? nullptr : std::dynamic_pointer_cast<BaseTablet>(res.value());
```
Comment (platoneko) [inline be/src/service/point_query_executor.cpp:198]: 这里不需要对指针类型进行转换，另外可以使用 `DORIS_TRY` 宏简化写法

Comment (platoneko) [reply be/src/service/point_query_executor.cpp:198]: 如果用 `ExecEnv::get_tablet` 返回的不是 error 则不会有 `tablet == nullptr` 的情况，可以删掉下面的判断

**Review by platoneko — CHANGES_REQUESTED** [review]
Please resolve the comment

File: be/src/service/point_query_executor.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/service/point_query_executor.cpp]: done

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

File: be/src/service/point_query_executor.cpp [hunk 2]
```diff
@@ -267,6 +266,10 @@ Status PointQueryExecutor::_lookup_row_key() {
     SCOPED_TIMER(&_profile_metrics.lookup_key_ns);
     // 2. lookup row location
     Status st;
+    if (_version >= 0) {
+        CHECK(config::is_cloud_mode()) << "Only cloud mode support snapshot read at present";
+        RETURN_IF_ERROR(std::dynamic_pointer_cast<CloudTablet>(_tablet)->sync_rowsets(_version));
```
Comment (xiaokang) [inline be/src/service/point_query_executor.cpp:271]: Does `_tablet->sync_rowsets` has side effect on _tablet to influnce other queries?

**Review by platoneko — APPROVED** [review]

File: be/src/service/point_query_executor.cpp (hunk 2 above)
Comment (eldenmoon) [reply be/src/service/point_query_executor.cpp:271]: no side effect

**Pull request merged**